driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTest.java

```diff
@@ -152,11 +151,15 @@ private void assertOutcome() {
 
     private void assertOperation(final BsonDocument operation) {
         OperationResult result = executeOperation(operation);
+        context.push(ContextElement.ofOperation(operation, result));
         if (operation.containsKey("expectResult")) {
-            valueMatcher.assertValuesMatch(operation.get("expectResult"), requireNonNull(result.getResult()));
+            assertNotNull(context.getMessage("The operation expects a result but an exception occurred"), result.getResult());
```

Comment:
This change turns an NPE into an assertion, which is correct, as there are cases where the test fails because it expect a result and the driver threw instead